src/binfmt_misc.jl

```diff
@@ -0,0 +1,232 @@
+malformed, or uses unsupported features) it an `ArgumentError` will be thrown.
+"""
+function BinFmtRegistration(file::String)
+    enabled = false
+    interpreter = nothing
+    flags = nothing
+    offset = nothing
+    magic = nothing
+    mask = nothing
+    for l in strip.(filter(!isempty, split(String(read(file)), "\n")))
+        # Handle enabled/disabled line
+        if l in ("enabled", "disabled")
+            enabled = l == "enabled"
+        elseif startswith(l, "interpreter ")
+            interpreter = l[13:end]
+        elseif startswith(l, "flags:")
+            flags = l[8:end]
+        elseif startswith(l, "offset ")
+            offset = parse(Int64, l[8:end])
+        elseif startswith(l, "magic ")
+            magic = hex2bytes(l[7:end])
+        elseif startswith(l, "mask ")
+            mask = hex2bytes(l[6:end])
+        else
+            @warn("Unknown `binfmt_misc` configuration directive", line=l)
+        end
+    end
+
+    # Ensure we are only dealing with properly fully-specified binfmt_misc registrations
+    @check_specified interpreter
+    @check_specified flags
+    @check_specified offset
+    @check_specified magic
+
+    # If we found a disabled binfmt_misc registration, just ignore it
+    if !enabled
+        return nothing
+    end
+    
+    return BinFmtRegistration(basename(file), interpreter, flags, offset, magic, mask)
+end
+
+function formats_match(a::BinFmtRegistration, b::BinFmtRegistration)
+    return (a.magic .& a.mask) == (b.magic .& b.mask)
+end
+
+
+"""
+    read_binfmt_misc_registrations()
+
+Return a list of `BinFmtRegistration` objects, one per readable registration, as found
+sitting in `/proc/sys/fs/binfmt_misc/*`.  Registrations that cannot be parsed are
+silently ignored.
+"""
+function read_binfmt_misc_registrations()
+    if !check_binfmt_misc_loaded()
+        return String[]
+    end
+
+    registrations = BinFmtRegistration[]
+    for f in readdir("/proc/sys/fs/binfmt_misc"; join=true)
+        # Skip "special" files
+        if basename(f) ∈ ("register", "status")
+            continue
+        end
+
+        try
+            reg = BinFmtRegistration(f)
+            if reg !== nothing
+                push!(registrations, reg)
+            end
+        catch e
+            if isa(e, ArgumentError)
+                continue
+            end
+            rethrow(e)
+        end
+    end
+    return registrations
+end
+
+"""
+    write_binfmt_misc_registration(reg::BinFmtRegistration)
+
+Write a `binfmt_misc` registration out to the kernel's `register` file endpoint.
+Requires `sudo` privileges.
+"""
+function write_binfmt_misc_registration(reg::BinFmtRegistration)
+    try
+        open(`$(sudo_cmd()) tee -a /proc/sys/fs/binfmt_misc/register`, write=true) do io
+            write(io, register_string(reg))
+        end
+    catch e
+        @error("Unable to register binfmt_misc format", register_string=register_string(reg))
+        rethrow(e)
```

Comment:
```suggestion
       # rethrow(e)
```

### Motivation

Suppose we have the situation in which:
1. Some of the registrations already exist on disk in the user's system.
2. The user does not have `sudo` access.

Since some of the formats are already registered, the user might still be able to run the desired binaries. So I don't think that a lack of `sudo` should necessarily be a fatal error.

Reply:
1) We only try to register things that aren't already registered.
2) I figure that if someone asks for `multiarch` support, it should be fatal if we can't set it up properly.  Perhaps we should make it more granular, asking only for the architecture that we're interested in?

Reply:
> I figure that if someone asks for multiarch support, it should be fatal if we can't set it up properly. Perhaps we should make it more granular, asking only for the architecture that we're interested in?

Hmmm. How about, by default, we try to set up all architectures, and throw a fatal error if we cannot register everything.

But we can have an optional keyword argument, and if the user provides a list of architectures, then we only try to set up that list.

Reply:
Yeah, right now it's being set by the `SandboxConfig` having `multiarch = true` set.  Perhaps we can instead have it be `multiarch = [:aarch64, :x86_64]` or something like that?

Reply:
SGTM.

And I think I agree that by default, we make the list include all the arches.

Reply:
I'm not sure about that one actually... I think not doing anything with multiarch stuff is the best default.  I think multiarch support should be opt-in, and if you're opting in, you should know what arches you need.  For BB, for instance, we're going to want them all, but for other usecases, we probably only want one.

Reply:
> if you're opting in, you should know what arches you need.

Ah, that's a good point.

So what should the default value be? An empty vector?